tests/test_dsl.py

```diff
@@ -320,3 +327,321 @@ def assert_cohorts_equivalent(dsl_cohort, qm_cohort):
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_intseries_lt():
+    series1 = IntSeries(ValueFromRow(source=None, column="number"))
+    series2 = IntSeries(ValueFromRow(source=None, column="number"))
+
+    output = series1 < series2
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.operator == "__lt__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_intseries_le():
+    series1 = IntSeries(ValueFromRow(source=None, column="number"))
+    series2 = IntSeries(ValueFromRow(source=None, column="number"))
+
+    output = series1 <= series2
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.operator == "__le__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_intseries_ne():
+    series1 = IntSeries(ValueFromRow(source=None, column="number"))
+    series2 = IntSeries(ValueFromRow(source=None, column="number"))
+    output = series1 != series2
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.operator == "__ne__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_patientseries_eq():
+    output = PatientSeries(value=Value()) == PatientSeries(value=Value())
+
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.lhs == ""
+    assert output.value.operator == "__eq__"
+
+    assert isinstance(output.value.rhs, PatientSeries)
+    assert output.value.rhs.value == ""
+
+
+def test_patientseries_is_comparator():
+    assert not PatientSeries(value=Value())._is_comparator()
+
+    assert PatientSeries(value=Comparator())._is_comparator()
```

Comment:
Why do we need to test this private method?

Reply:
It wasn't being tested so I thought I'd bump up code coverage!  However, I felt it was useful to have it tested too as it's a core part of the lower level DSL machinery, eg how the base `__invert__` works.

Reply:
If it's not being tested then what's it for?  Why not delete it?

Reply:
I just traced this down… in the `DateSeries.round_to_first_*` methods we pass in the relevant ValueFromFunction subclasses to `DateSeries`, which are subclasses of Value.  The `_is_comparator` method is then used in `dsl.categorise` to decide if we should remove nulls from the value.

So it's definitely in use, but DateSeries haven't been used in the categorise tests.

Reply:
Sounds like we're missing a test for `categorise`?

Reply:
Good point.  Although I think it's important to also keep a test like this around so any errors in `_is_comparator()` are clear!

Reply:
I disagree -- I can't see any circumstances under which we learn anything useful from the test failing.

It'll only fail if we change the implementation, and then whoever changes the implementation will have to work out whether it was a useful test or not.

Reply:
Yeah, I agree with Peter here.